torch/distributed/_checkpoint/storage_writer.py

```diff
@@ -0,0 +1,72 @@
+from .metadata import Metadata, ReadWriteRequest
+from typing import Dict
+import os
+from torch.futures import Future
+import pickle
+import torch.distributed as dist
+import abc
+
+class StorageWriter(abc.ABC):
+    """
+    Interface to write to underlying storage system
+    """
+    @abc.abstractmethod
+    def write(self, req: ReadWriteRequest) -> Future:
+        """
+        Performs a write request and returns a Future to wait on.
+        Args:
+            req (ReadWriteRequest): see `./metadata.py`
+        """
+        pass
+
+    @abc.abstractmethod
+    def write_metadata(self, metadata: Metadata) -> None:
```

Comment:
qq, why write() is async but write_metadata is sync()? Are they written to different destinations?